samples: add i2c-target samples

bravl opened the pull request:
Add two simple i2c target samples

1. custom i2c target: sample how to use the i2c target api
2. i2c eeprom target: sample how to initialize and use the i2c driver target API

bravl committed:
samples: drivers: add i2c eeprom target sample

Add simple sample for the i2c target driver interface using the
i2c-target-eeprom as a target device.

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>

bravl committed:
samples: drivers: i2c: add i2c target api sample

Add simple sample which demonstrates how to setup a custom i2c target.

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>

kartben commented on samples/drivers/i2c/target_eeprom/README.rst:
```diff
@@ -0,0 +1,31 @@
+********
+
+This sample demonstrates how to setup and use the I2C target driver API using the
+`zephyr,i2c-target-eeprom` device.
```
```suggestion
:dtcompatible:`zephyr,i2c-target-eeprom` device.
```

bravl replied:
Fixed!

kartben commented on samples/drivers/i2c/target_eeprom/README.rst:
```diff
@@ -0,0 +1,31 @@
+Overview
+********
+
+This sample demonstrates how to setup and use the I2C target driver API using the
```
```suggestion
This sample demonstrates how to setup and use the :ref:`i2c-target-api` using the
```

bravl replied:
Fixed!

kartben commented on samples/drivers/i2c/custom_target/README.rst:
```diff
@@ -0,0 +1,30 @@
+Overview
+********
+
+This sample demonstrates how to setup an I2C custom target on the I2C interface.
```
```suggestion
This sample demonstrates how to setup an I2C custom target on the I2C interface
using the :ref:`i2c-target-api`.
```

bravl replied:
Fixed!

kartben reviewed (changes requested):
cool, thanks for this! 
Some minor suggested documentation fixes

teburd reviewed (dismissed):
I like it, thanks for adding this!

kartben reviewed (changes requested):
actually, one more request for change: can you please update the MAINTAINERS file to add the new `samples/drivers/i2c` folder to the "Drivers: I2C" entry? This way future updates/additions to the I2C samples will be correctly assigned to @teburd 😛

bravl committed:
MAINTAINERS.yml: add new samples/drivers/i2c

Add samples/drivers/i2c to the maintainers file so the correct code
owners get added in PRs

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>

bravl commented:
> actually, one more request for change: can you please update the MAINTAINERS file to add the new `samples/drivers/i2c` folder to the "Drivers: I2C" entry? This way future updates/additions to the I2C samples will be correctly assigned to @teburd 😛

added to Maintainers.yml :+1:

kartben reviewed (approved):
Thanks, @bravl!
